it/v2/endpoints/CreateHistoricNonFhlUkPiePeriodSummaryControllerISpec.scala

```diff
@@ -0,0 +1,430 @@
+
+      "any valid consolidated request is made" in new Test {
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+          DownstreamStub.onSuccess(DownstreamStub.POST, ifsUri, Status.OK, ifsResponse)
+
+        }
+
+        val response: WSResponse = await(request().post(requestBodyJsonConsolidatedExpense))
+        response.status shouldBe Status.CREATED
+        response.json shouldBe responseBody
+        response.header("Content-Type") shouldBe Some("application/json")
+      }
+    }
+
+    "return bad request error" when {
+      "badly formed json body is submitted" in new Test {
+        private val json =
+          s"""
+             |{
+             | badJson
+             | }
+           """.stripMargin
+
+        override def setupStubs(): StubMapping = {
+          AuditStub.audit()
+          AuthStub.authorised()
+          MtdIdLookupStub.ninoFound(nino)
+        }
+        val response: WSResponse = await(request().addHttpHeaders(("Content-Type", "application/json")).post(json))
+        response.status shouldBe Status.BAD_REQUEST
+        response.json shouldBe Json.toJson(BadRequestError)
+      }
+    }
+
+    "return error according to spec" when {
+      "validation error" when {
+        def validationErrorTest(requestNino: String, requestBody: JsValue, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"validation fails with ${expectedBody.code} error" in new Test {
+
+            override val nino: String = requestNino
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+            }
+
+            val response: WSResponse = await(request().post(requestBody))
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          ("AA1123A", requestBodyJson, Status.BAD_REQUEST, NinoFormatError),
+          ("AA123456A", invalidToDateRequestJson, Status.BAD_REQUEST, ToDateFormatError),
+          ("AA123456A", invalidFromDateRequestJson, Status.BAD_REQUEST, FromDateFormatError),
+          ("AA123456A", bothExpensesSuppliedRequestJson, Status.BAD_REQUEST, RuleBothExpensesSuppliedError.copy(paths=Some(Seq("/expenses/consolidatedExpenses")))),
+          ("AA123456A", missingFromAndToDateRequestJson, Status.BAD_REQUEST,  RuleIncorrectOrEmptyBodyError.copy(paths=Some(Seq("/fromDate", "/toDate")))),
+          ("AA123456A", toDateBeforeFromDateRequestJson, Status.BAD_REQUEST, RuleToDateBeforeFromDateError),
+          ("AA123456A", invalidValueRequestJson, Status.BAD_REQUEST, allInvalidValueRequestError)
+        ) //Should we add a test for rentARoom.amountClaimed and rentARoom.rentsReceived being missing? Tech spec says they're mandatory, business spec says they're optional.
```

Comment:
Code has them as optional - so presumably in agreement with RAML. 

I believe the backend values are optional and there was a question over whether we should make them mandatory, but it looks like we've done what we did with the similar non-historic endpoints. So I suspect the tech spec mappings need adjusting.

Reply:
Removed the comment